src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -1022,35 +1031,36 @@ public string ToStringClass()
             // This is important because if the last range ends in LastChar, we won't append
             // LastChar to the list.
             int rangeLen = _rangelist.Count * 2;
-            StringBuilder sb = StringBuilderCache.Acquire(rangeLen + _categories.Length + 3);
+            int strGuessCount = rangeLen + _categories.Length + 3;
 
-            int flags;
-            if (_negate)
-                flags = 1;
-            else
-                flags = 0;
+            Span<char> buffer = strGuessCount <= 256 ? stackalloc char[256] : null;
+            ValueStringBuilder vsb = buffer != null ?
+                new ValueStringBuilder(buffer) :
+                new ValueStringBuilder(strGuessCount);
+
+            int flags = _negate ? 1 : 0;
 
-            sb.Append((char)flags);
-            sb.Append((char)rangeLen);
-            sb.Append((char)_categories.Length);
+            vsb.Append((char)flags);
+            vsb.Append((char)rangeLen);
+            vsb.Append((char)_categories.Length);
 
             for (int i = 0; i < _rangelist.Count; i++)
             {
                 SingleRange currentRange = _rangelist[i];
-                sb.Append(currentRange.First);
+                vsb.Append(currentRange.First);
 
                 if (currentRange.Last != LastChar)
-                    sb.Append((char)(currentRange.Last + 1));
+                    vsb.Append((char)(currentRange.Last + 1));
             }
 
-            sb[SETLENGTH] = (char)(sb.Length - SETSTART);
+            vsb[SETLENGTH] = (char)(vsb.Length - SETSTART);
 
-            sb.Append(_categories);
+            vsb.Append(_categories.ToString());
```

Comment:
You can use the new `GetChunks` member to avoid this ToString(), e.g.
```C#
foreach (ReadOnlyMemory<char> chunk in _categories.GetChunks())
{
    vsb.Append(chunk.Span);
}
```
or potentially:
```C#
Span<char> categories = vsb.AppendSpan(_categories.Length);
foreach (ReadOnlyMemory<char> chunk in _categories.GetChunks())
{
    chunk.CopyTo(categories);
    categories = categories.Slice(chunk.Length);
}
```
I'm not sure which would be faster, or if it matters.  The first is simpler and probably better, in particular for the (common?) case where the StringBuilder contains a single chunk.

Reply:
thanks a lot 👍